### PR TITLE
core-components: Remove CheckboxTree component (used in Table Filters)

### PR DESCRIPTION
**Pull request by ainhoaL**
Signed-off-by: Ainhoa Larumbe <ainhoal@spotify.com>

DO NOT MERGE until right before the release on September 30th

## Hey, I just made a Pull Request!

For issue #6618 
Removed the option for having CheckboxTree as a Table filter:
- Deleted the `CheckboxTree` component and its uses in `Filters`
- Removed the filter type `'checkbox-tree'` from the `TableFilter` types.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/red-hats-walk.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/core-components': minor
+---
+
+Checkbox tree filters are no longer available in the Table component:
+
+- Deleted the `CheckboxTree` component
+- Removed the filter type `'checkbox-tree'` from the `TableFilter` types.
```
Comment (Rugvip) [inline .changeset/red-hats-walk.md:8]: So `core-components` is at [stability index 1](https://backstage.io/docs/overview/stability-index#core-components-githubhttpsgithubcombackstagebackstagetreemasterpackagescore-components), which means this is fine with just a guide. But tbh I think we should be a bit nicer here and not break stuff immediately.

In this case that'd mean that we keep this PR around and set a merge date of it in two weeks or more, and then only merge and release a deprecation of this for now. The `TableFilter` would print a deprecation warning if `'checkbox-tree'` is used, and we mark the `CheckboxTree` component with `@deprecated` and print a warning there as well.

**Review by Rugvip — COMMENTED** [review]
Thank you for tidying things up! :grin:

Just some thoughts on proceeding a bit more slowly with the change

File: .changeset/red-hats-walk.md (hunk 1 above)
Comment (ainhoaL) [reply .changeset/red-hats-walk.md:8]: That is a fair comment! You are right, the nice thing to do would be to deprecate them first, so I will do as you suggest. Let's say we will merge this one 30/09.

**Comment by freben** [comment]
@ainhoaL So! Now that the deprecation PR is merged, this one will need a rebase unfortunately :)

**Review by camilaibs — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]
Let's :shipit:

**Pull request merged**